src/app/pages/transactions/in-progress/transactions-in-progress-table-data-source.ts

```diff
@@ -210,6 +213,9 @@ export class TransactionsInProgressTableDataSource extends TableDataSource<Trans
           (actionDef as TableViewTransactionActionDef).action(transaction, this.dialog, this.refreshData.bind(this));
         }
         break;
+      case ChargingStationButtonAction.CHECK_CHARGING_PLANS:
+        this.checkChargingPlansAction.action('charging-stations#chargingplans?ChargingStationID=' + transaction.chargeBoxID + '&TransactionID=' + transaction.id);
```

Comment:
Why is there no downcast here?

Reply:
Downcast as to pass the object transaction to the action and handle it in button file ?

Reply:
I would expect that kind of call:

`(actionDef as TableViewTransactionActionDef).action(transaction, this.dialog, this.refreshData.bind(this));`

Reply:
Ok

Reply:
Done.